packages/expo-notifications/ios/EXNotifications/Notifications/EXNotificationCenterDelegate.m

```diff
@@ -161,9 +161,13 @@ - (void)addDelegate:(id<EXNotificationsDelegate>)delegate
   [_delegates addPointer:(__bridge void * _Nullable)(delegate)];
   if ([delegate respondsToSelector:@selector(userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:)]) {
     UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
-    for (UNNotificationResponse *response in _pendingNotificationResponses) {
+    NSArray *responsesToProcess = [_pendingNotificationResponses copy];
+
+    for (UNNotificationResponse *response in responsesToProcess) {
       [delegate userNotificationCenter:center didReceiveNotificationResponse:response withCompletionHandler:^{
-        // completion handler doesn't need to do anything
+        @synchronized (self) {
```

Comment:
this is too complicated. If we're making a copy of `_pendingNotificationResponses`, why can't we make the copy, then clear the `_pendingNotificationResponses`, and then iterate over the copy?